### PR TITLE
(fix) O3-5588: Correctly apply Login Location tag filter in useLocationCount

### PR DESCRIPTION
**Pull request by sudeshkar**
# Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-docs.openmrs.org/docs/frontend-modules/contributing.en-US#contributing-guidelines) label. See existing PR titles for inspiration.
- [ ] My work is based on designs, which are linked or shown either in the Jira ticket or the description below. (See also: [Styleguide](http://om.rs/o3ui))
- [x] My work includes tests or is validated by existing tests.
- [ ] I have updated the esm-framework and storybook mocks ([mock.tsx](https://github.com/openmrs/openmrs-esm-core/blob/main/packages/framework/esm-framework/mock.tsx), [mock-jest.tsx](https://github.com/openmrs/openmrs-esm-core/blob/main/packages/framework/esm-framework/mock-jest.tsx), and [storybook mocks](https://github.com/openmrs/openmrs-esm-core/blob/main/packages/tooling/storybook/mocks/)) to reflect any API changes.

## Summary

`useLocationCount` in `location-picker.resource.ts` used `String.prototype.concat()` to append the `_tag=Login Location` FHIR filter when `useLoginLocationTag` was enabled.  However, `concat()` returns a new string without mutating the original, so the tag  filter was silently never applied regardless of the config value.

This meant any deployment using login-location tagging was unknowingly counting all locations instead of only tagged ones. Because `chooseLocation.useLoginLocationTag` defaults to `true`, this affected the default configuration, not only sites that explicitly opted in. As a result, `locationCount` returned the total location count rather than the tagged count, which broke the auto-redirect logic that skips the location picker when only one valid login location exists.

**Changes made:**
- Replaced `url.concat()` with `URLSearchParams` which correctly builds and encodes 
  the query string, consistent with `useLoginLocations()` in `login.resource.ts`
- Corrected FHIR parameter name from `tag=` to `_tag=` which is the correct FHIR R4 
  parameter name
- Added unit tests for `useLocationCount` covering URL construction for both flag 
  values, response shape, and the null `firstLocation` edge case

## Screenshots

No UI changes. The fix corrects a silent data-fetching bug. The location picker screen behaviour changes only for sites using login-location tagging (the default),  where the correct tagged location count is now returned from the server.

## Related Issue

Related Issue: https://openmrs.atlassian.net/browse/O3-5588

### CODE REVIEW

File: packages/apps/esm-login-app/src/location-picker/location-picker.resource.test.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        }) as FetchResponse<any>,
+    );
+
+    const { result, unmount } = renderHook(() => useLocationCount(false));
+    unmount();
+
+    const entry = undefined;
+    const firstLocation = entry ? entry[0] : null;
+    expect(firstLocation).toBeNull();
+  });
```
[the remark this answers is not in the document]
Comment (sudeshkar) [reply packages/apps/esm-login-app/src/location-picker/location-picker.resource.test.ts]: Fixed the issue could you please review and give me a feedback

File: packages/apps/esm-login-app/src/location-picker/location-picker.resource.test.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
+import { renderHook, waitFor } from '@testing-library/react';
+import { beforeEach, describe, expect, it, vi } from 'vitest';
+import { openmrsFetch, type FetchResponse } from '@openmrs/esm-framework';
+import { mockLoginLocations, mockSoleLoginLocation } from '../../__mocks__/locations.mock';
+import { useLocationCount } from './location-picker.resource';
+import { SWRConfig } from 'swr';
+import React from 'react';
+
+const mockOpenmrsFetch = vi.mocked(openmrsFetch);
+const wrapper = ({ children }) => React.createElement(SWRConfig, { value: { provider: () => new Map() } }, children);
```
[the remark this answers is not in the document]
Comment (sudeshkar) [reply packages/apps/esm-login-app/src/location-picker/location-picker.resource.test.ts:13]: @denniskigen  Done — added the wrapper to all renderHook calls so every test gets 
a fresh SWR cache. Thanks for the suggestion!

**Comment by sudeshkar** [comment]
Done — added the `wrapper` to all `renderHook` calls so every test now 
gets a fresh SWR cache. All 40 tests in the package pass. 
image

@denniskigen

File: packages/apps/esm-login-app/src/location-picker/location-picker.resource.ts [hunk 3]
```diff
@@ -86,10 +86,11 @@ export function useDefaultLocation(isUpdateFlow: boolean) {
...
 
 export function useLocationCount(useLoginLocationTag: boolean) {
-  const url = `/ws/fhir2/R4/Location?_count=1`;
+  const urlParams = new URLSearchParams({ _count: '1' });
   if (useLoginLocationTag) {
-    url.concat(`&tag=Login Location`);
+    urlParams.append('_tag', 'Login Location');
   }
+  const url = `/ws/fhir2/R4/Location?${urlParams.toString()}`;
   const { data, error, isLoading } = useSwrImmutable<FetchResponse<LocationResponse>>(url, openmrsFetch, {
```
[the remark this answers is not in the document]
Comment (ibacher) [reply packages/apps/esm-login-app/src/location-picker/location-picker.resource.ts:101]: It was suggesting that you use something like:

`${fhirBaseUrl}/Location?${urlParams}`

**Review by ibacher — APPROVED** [review]

**Comment by sudeshkar** [comment]
@ibacher Done — switched to fhirBaseUrl. Thanks for the suggestion!
image
@denniskigen
please review and give me you feedbacks it means a lot

**Comment by sudeshkar** [comment]
@ibacher @denniskigen Kindly review this and give me the feedback

File: packages/apps/esm-login-app/src/location-picker/location-picker.resource.ts [hunk 4]
```diff
@@ -86,10 +93,11 @@ export function useDefaultLocation(isUpdateFlow: boolean) {
 }
 
 export function useLocationCount(useLoginLocationTag: boolean) {
-  const url = `/ws/fhir2/R4/Location?_count=1`;
+  const urlParams = new URLSearchParams({ _count: '1' });
   if (useLoginLocationTag) {
-    url.concat(`&tag=Login Location`);
+    urlParams.append('_tag', 'Login Location');
```
Comment (denniskigen) [inline packages/apps/esm-login-app/src/location-picker/location-picker.resource.ts:98]: This is the heart of the fix, and it is correct. Two things were broken here before: `url.concat(...)` returned a new string that was discarded, so the tag was never applied, and the parameter was `tag=` rather than the FHIR R4 `_tag=`. Switching to `URLSearchParams` fixes both, and it now matches how `useLoginLocations` builds the same query in `login.resource.ts`. Nicely done.

**Review by denniskigen — APPROVED** [review]
LGTM. Thanks, @sudeshkar!
